src/app/cake/cake.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Cake } from "./cake.class";
+
+let counter = 1;
+const data: Cake[] = [];
+
+/**
+ * Simulate a slow db connection based on the number of items in the list
+ * @param val
+ */
+function simulate<T>(val: T){
+    return new Promise<T>((resolve)=>{
+        setTimeout(()=>{
+            resolve(val);
+        }, data.length * 100);
+    });
+}
+export const cakeService = {
```

Comment:
Don't you think it will be a better idea to export a single function instead of export a single object with functions? 

As a good practice is always better to export and import a single element from a file instead of the hole package.

Reply:
Yes, i have my reasons i suppose, I like to do that because it helps the intellisense things in typescript. It is easier to type "cakeS" to auto complete and import the cake service and see the methods and vars implemented that way, otherwise you should be looking the file and see what is in there or using the intellisense the import statement but that's a more manual approach.